### DISCUSSION TITLE
v5.21.0

### DISCUSSION THREAD

**Discussion by richardwilkes** [richardwilkes/gcs#829] [Announcements]
**New Features**

- Added a "Names Only" checkbox to the search/filter fields to only work against the name of the item and not all of the displayed information (#546)
- Added a "Jump to Search/Filter Field" menu item and key binding to focus the search/filter field (#804)
- Added support for custom Self-Control Roll values filling in the missing spots between CR6 and CR15 (#822)
- Added the ability to check for a tag when using the equipped equipment prerequisite (#827)
- An **EXPERIMENTAL** (and **INCOMPLETE**) built-in web server has been added (configured under Settings > Web Server Settings, off by default) that allows users to view character sheets via a browser. At this point in time, only very limited editing is supported. Depending upon feedback, this particular feature may or may not be kept and expanded upon. Note that the usual caveats about exposing your machine to the internet apply with this feature, so only enable it if you're comfortable with the potential risks.

**Bug Fixes**

- Fixed the inability to make a selection in a dropdown menu using the keyboard on Windows & Linux (#802)
- Parry and Block bonuses should now show up in tooltips (#808)
- Fragmentation damage with an armor divisor of 1 no longer displays the armor divisor (#812)
- Fixed a case where lists would jump back to a previous scroll position when they shouldn't have (#815)
- Fixed a bug that would scramble the display of the points record panel after deleting a row (#818)
- Fixed an inaccurate tooltip for Parry using an unbalanced weapon
- Added a 1 second timeout to document handoff operations to prevent potentially endless waiting

<hr /><em>This discussion was created from the release <a href='https://github.com/richardwilkes/gcs/releases/tag/v5.21.0'>v5.21.0</a>.</em>

**Comment by CReimer** [comment]
Some initial feedback from my side about the web server feature.
This is amazing. Please keep it around and expand it to be on par with the normal GUI.
I'm installing this on my webserver right now to have my character sheet available on my iPad for my next session.

This might not be what you originally intended, but the web interface could replace the unison GUI a some point.
Something like Discord does with Electron or like Spotify does with libCEF

**Reply by richardwilkes** [reply]
Thank you for the feedback. I've had a very small handful of people say they believe its worthwhile/useful, so I'm not sure whether I should really be investing any more time into it, especially given how much I dislike web development.

Like you, though, I've found it useful for being able to pull GCS sheets up on iPads.

**Reply by CReimer** [reply]
I am a web developer (mainly React/MobX based). I haven't looked at what framework you use, but if you need help with something specific, maybe you could collect that somewhere. I'm not opposed to to contributing code from time to time, if that means keeping the server feature.

I've even started to implement a web-based solution for myself (with the intend to open source it down the line). But at some point I failed due to the complexity of the calculations and my limited free time.

**Reply by richardwilkes** [reply]
I'm using Svelte (v4) + Typescript at the moment.

**Reply by CReimer** [reply]
I took a quick look at their documentation. It reminds me of React in many places and something like MobX (MST) is seemingly already directly integrated there. I would be fine with that.

Typescript is also no problem. From my point of view, there should only be Typescript and JavaScript should disappear.

**Comment by jutis** [comment]
The webserver is great, Hoping to see more development of it in the future, maybe even integrate die rolling for skills and other things as well?
Great job!!
